lib/rebaser-server/src/change_set_processor_task.rs

```diff
@@ -0,0 +1,420 @@
+        // Actions are **ONLY** ever dispatched from the default change set for a workspace.
+        if matches!(rebase_status, RebaseStatus::Success { .. }) {
+            // If we find dependent value roots, then notify the serial dvu task to run at least
+            // one more dvu
+            if ctx
+                .workspace_snapshot()?
+                .has_dependent_value_roots()
+                .await?
+            {
+                run_notify.notify_one();
+            }
+
+            if let Some(workspace) = Workspace::get_by_pk(&ctx, &workspace_id.into()).await? {
+                if workspace.default_change_set_id() == ctx.visibility().change_set_id {
+                    let mut change_set = ChangeSet::find(&ctx, ctx.visibility().change_set_id)
+                        .await?
+                        .ok_or(RebaseError::MissingChangeSet(change_set_id.into()))?;
+                    if WorkspaceSnapshot::dispatch_actions(&ctx).await? {
+                        // Write out the snapshot to get the new address/id.
+                        let new_snapshot_id = ctx
+                            .write_snapshot()
+                            .await?
+                            .ok_or(dal::WorkspaceSnapshotError::WorkspaceSnapshotNotWritten)?;
+                        // Manually update the pointer to the new address/id that reflects the new
+                        // Action states.
+                        change_set.update_pointer(&ctx, new_snapshot_id).await?;
+                        // No need to send the request over to the rebaser as we are the rebaser.
+                        ctx.commit_no_rebase().await?;
+                    }
+                }
+            }
+        }
+
+        // If a reply was requested, send it
+        if let Some(reply) = maybe_reply {
+            let response = EnqueueUpdatesResponse::new_current(EnqueueUpdatesResponseVCurrent {
+                id: request.id,
+                workspace_id: request.workspace_id,
+                change_set_id: request.change_set_id,
+                status: rebase_status,
+            });
+
+            let info = ContentInfo::from(&response);
+
+            // TODO(fnichol): add span propagation
+            let mut headers = HeaderMap::new();
+            info.inject_into_headers(&mut headers);
+
+            nats.publish_with_headers(reply, headers, response.to_vec()?.into())
+                .await
+                .map_err(Error::PublishReply)?;
+        }
+
+        // TODO(fnichol): hrm, is this *really* true that we've written to the change set. I mean,
+        // yes but until a dvu has finished this is an incomplete view?
+        let mut event = WsEvent::change_set_written(&ctx, change_set_id.into()).await?;
+        event.set_workspace_pk(workspace_id.into());
+        event.set_change_set_id(Some(change_set_id.into()));
+        event.publish_immediately(&ctx).await?;
```

Comment:
Smells like a "fuck it we ball, reload that UI" trigger. I'm okay with it for now, but you're right... not necessarily going to be true. Good to call out!